# Rafflesiaceae_phylogenomics
This repository contains scripts for the phylogenomic study of Rafflesiaceae. If you have any questions, feel free to reach out to Liming Cai daybreak.chua@gmail.com

Citation: Liming Cai, Liang Liu, Charles C Davis. The danger zone: the joint trap of incomplete lineage sorting and long-branch attraction in placing Rafflesiaceae. In review.

Subfolders:

**1_Orthogroup_aln_prep**: Script used to merge two phylogenomics datasets from Cai et al. (2019) and Cai et al. (2021) in order to investigate the placement of Rafflesiaceae.

**2_Alignment_geneTree**: MAFFT and IQTREE command line to generate sequence alingment and individual gene trees.

**3_gene_shopping**: SORTDATE command line to extract phylogenetic properties of individual genes.

**4_phylogeny_hypo_generation**: Command lines to infer species tree under the concatenation and coalescent models.

**5_GGI_test**: Application of gene genealogy interrogation (GGI) method from Arcila et al. (2017) to test alternative placement of Rafflesiaceae.

**6_hypothesis_testing**: Testing alternative placements of Rafflesiacaee under the concatenation and coalescent models.

**7_starbeast**: Inferring the placement of Rafflesiaceae under the full coalescent model in starBEAST with a reduced dataset.

**8_MSC_simulation**: Simulating sequence under the multi-species coalescent model using empirical parameters to test whether it is feasible to place Rafflesiaceae.

For specific steps, go to the README in each subfolders.
