setwd('Documents/GitHub/Rafflesiaceae_phylogenomics/5_GGI_test/')
x=read.table('AU_rank1.txt',row.names=1,col.names = paste0("V",seq_len(1459)), fill = TRUE)
pdf(file = 'AU.pdf',width = 4,height = 4)
plot(0,xlim=c(0,500),ylim=c(0.5,1),xlab='Gene number',ylab='AU test p-value')
y=sort(x['H2',],decreasing = T)
lines(1:length(y),y,col='black')
y=sort(x['H3',],decreasing = T)
lines(1:length(y),y,col='red')
y=sort(x['H4',],decreasing = T)
lines(1:length(y),y,col='blue')
y=sort(x['H7',],decreasing = T)
lines(1:length(y),y,col='green')
y=sort(x['H10',],decreasing = T)
lines(1:length(y),y,col='pink')
y=sort(x['H1',],decreasing = T)
lines(1:length(y),y,col='purple')
y=sort(x['H5',],decreasing = T)
lines(1:length(y),y,col='grey')
y=sort(x['H6',],decreasing = T)
lines(1:length(y),y,col='orange')
abline(h = 0.95,lty=2)
dev.off()

x=read.table('KH_rank1.txt',col.names = paste0("V",seq_len(500)), fill = TRUE,row.names=1)
pdf(file = 'KH.pdf',width = 4,height = 4)
plot(0,xlim=c(0,500),ylim=c(0.5,1),xlab='Gene number',ylab='KH test p-value')
y=sort(x['H2',],decreasing = T)
lines(1:length(y),y,col='black')
y=sort(x['H3',],decreasing = T)
lines(1:length(y),y,col='red')
y=sort(x['H4',],decreasing = T)
lines(1:length(y),y,col='blue')
y=sort(x['H7',],decreasing = T)
lines(1:length(y),y,col='green')
y=sort(x['H10',],decreasing = T)
lines(1:length(y),y,col='pink')
y=sort(x['H1',],decreasing = T)
lines(1:length(y),y,col='purple')
y=sort(x['H5',],decreasing = T)
lines(1:length(y),y,col='grey')
y=sort(x['H6',],decreasing = T)
lines(1:length(y),y,col='orange')
abline(h = 0.95,lty=2)
dev.off()

x=read.table('SH_rank1.txt',col.names = paste0("V",seq_len(500)), fill = TRUE,row.names=1)
pdf(file = 'SH.pdf',width = 4,height = 4)
plot(0,xlim=c(0,500),ylim=c(0.5,1),xlab='Gene number',ylab='SH test p-value')
y=sort(x['H2',],decreasing = T)
lines(1:length(y),y,col='black')
y=sort(x['H3',],decreasing = T)
lines(1:length(y),y,col='red')
y=sort(x['H4',],decreasing = T)
lines(1:length(y),y,col='blue')
y=sort(x['H7',],decreasing = T)
lines(1:length(y),y,col='green')
y=sort(x['H10',],decreasing = T)
lines(1:length(y),y,col='pink')
y=sort(x['H1',],decreasing = T)
lines(1:length(y),y,col='purple')
y=sort(x['H5',],decreasing = T)
lines(1:length(y),y,col='grey')
y=sort(x['H6',],decreasing = T)
lines(1:length(y),y,col='orange')
abline(h = 0.95,lty=2)
dev.off()

###################################
###################################
The following are misc old results, ignore

library(RColorBrewer)
#display.brewer.all()
cols <- brewer.pal(10, "Paired")
x=read.table('AU_sum.tsv',header = T)
plot(sort(x$H1,decreasing = T),pch=20,cex=0.4,type = 'l',xlab='Gene number',ylab='AU test p-value',col=cols[1],xlim=c(0,1420),ylim=c(0,1))
lines(sort(x$H2,decreasing = T),col=cols[2])
lines(sort(x$H4,decreasing = T),col=cols[3])
lines(sort(x$H5,decreasing = T),col=cols[4])
lines(sort(x$H7,decreasing = T),col=cols[5])
lines(sort(x$H9,decreasing = T),col=cols[6])
lines(sort(x$H12,decreasing = T),col=cols[7])
lines(sort(x$H15,decreasing = T),col=cols[8])
lines(sort(x$H16,decreasing = T),col=cols[9])
lines(sort(x$H17,decreasing = T),col=cols[10])

abline(h = 0.95,lty=2)

x=read.table('PP_sum.tsv',header = T)
plot(sort(x$H1,decreasing = T),pch=20,cex=0.4,type = 'l',xlab='Gene number',ylab='BIC posterior probability',col=cols[1],xlim=c(0,1420),ylim=c(0,1))
lines(sort(x$H2,decreasing = T),col=cols[2])
lines(sort(x$H4,decreasing = T),col=cols[3])
lines(sort(x$H5,decreasing = T),col=cols[4])
lines(sort(x$H7,decreasing = T),col=cols[5])
lines(sort(x$H9,decreasing = T),col=cols[6])
lines(sort(x$H12,decreasing = T),col=cols[7])
lines(sort(x$H15,decreasing = T),col=cols[8])
lines(sort(x$H16,decreasing = T),col=cols[9])
lines(sort(x$H17,decreasing = T),col=cols[10])
abline(h = 0.95,lty=2)


x=read.table('SH_sum.tsv',header = T)
plot(sort(x$H1,decreasing = T),pch=20,cex=0.4,type = 'l',xlab='Gene number',ylab='SH test p-value',col=cols[1],xlim=c(0,1420),ylim=c(0,1))
lines(sort(x$H2,decreasing = T),col=cols[2])
lines(sort(x$H4,decreasing = T),col=cols[3])
lines(sort(x$H5,decreasing = T),col=cols[4])
lines(sort(x$H7,decreasing = T),col=cols[5])
lines(sort(x$H9,decreasing = T),col=cols[6])
lines(sort(x$H12,decreasing = T),col=cols[7])
lines(sort(x$H15,decreasing = T),col=cols[8])
lines(sort(x$H16,decreasing = T),col=cols[9])
lines(sort(x$H17,decreasing = T),col=cols[10])
abline(h = 0.95,lty=2)

x=read.table('KH_sum.tsv',header = T)
plot(sort(x$H1,decreasing = T),pch=20,cex=0.4,type = 'l',xlab='Gene number',ylab='KH test p-value',col=cols[1],xlim=c(0,1420),ylim=c(0,1))
lines(sort(x$H2,decreasing = T),col=cols[2])
lines(sort(x$H4,decreasing = T),col=cols[3])
lines(sort(x$H5,decreasing = T),col=cols[4])
lines(sort(x$H7,decreasing = T),col=cols[5])
lines(sort(x$H9,decreasing = T),col=cols[6])
lines(sort(x$H12,decreasing = T),col=cols[7])
lines(sort(x$H15,decreasing = T),col=cols[8])
lines(sort(x$H16,decreasing = T),col=cols[9])
lines(sort(x$H17,decreasing = T),col=cols[10])
abline(h = 0.95,lty=2)

x=read.table('AU_sum.includePandaceae.tsv',header = T)
plot(sort(x$H1,decreasing = T),pch=20,cex=0.4,type = 'l',xlab='Gene number',ylab='AU test p-value',col=cols[1],xlim=c(0,200),ylim=c(0,1))
lines(sort(x$H2,decreasing = T),col=cols[2])
lines(sort(x$H4,decreasing = T),col=cols[3])
lines(sort(x$H5,decreasing = T),col=cols[4])
lines(sort(x$H7,decreasing = T),col=cols[5])
lines(sort(x$H9,decreasing = T),col=cols[6])
lines(sort(x$H12,decreasing = T),col=cols[7])
lines(sort(x$H15,decreasing = T),col=cols[8])
lines(sort(x$H16,decreasing = T),col=cols[9])
lines(sort(x$H17,decreasing = T),col=cols[10])

cols2=brewer.pal(7, "Dark2")
lines(sort(x$H3,decreasing = T),col=cols2[1])
lines(sort(x$H6,decreasing = T),col=cols2[2])
lines(sort(x$H8,decreasing = T),col=cols2[3])
lines(sort(x$H10,decreasing = T),col=cols2[4])
lines(sort(x$H11,decreasing = T),col=cols2[5])
lines(sort(x$H13,decreasing = T),col=cols2[6])
lines(sort(x$H14,decreasing = T),col=cols2[7])
abline(h = 0.95,lty=2)


x=read.table('PP_sum.includePandaceae.tsv',header = T)
plot(sort(x$H1,decreasing = T),pch=20,cex=0.4,type = 'l',xlab='Gene number',ylab='BIC posterior probability',col=cols[1],xlim=c(0,200),ylim=c(0,1))
lines(sort(x$H2,decreasing = T),col=cols[2])
lines(sort(x$H4,decreasing = T),col=cols[3])
lines(sort(x$H5,decreasing = T),col=cols[4])
lines(sort(x$H7,decreasing = T),col=cols[5])
lines(sort(x$H9,decreasing = T),col=cols[6])
lines(sort(x$H12,decreasing = T),col=cols[7])
lines(sort(x$H15,decreasing = T),col=cols[8])
lines(sort(x$H16,decreasing = T),col=cols[9])
lines(sort(x$H17,decreasing = T),col=cols[10])

cols2=brewer.pal(7, "Dark2")
lines(sort(x$H3,decreasing = T),col=cols2[1])
lines(sort(x$H6,decreasing = T),col=cols2[2])
lines(sort(x$H8,decreasing = T),col=cols2[3])
lines(sort(x$H10,decreasing = T),col=cols2[4])
lines(sort(x$H11,decreasing = T),col=cols2[5])
lines(sort(x$H13,decreasing = T),col=cols2[6])
lines(sort(x$H14,decreasing = T),col=cols2[7])
abline(h = 0.95,lty=2)

x=read.table('SH_sum.includePandaceae.tsv',header = T)
plot(sort(x$H1,decreasing = T),pch=20,cex=0.4,type = 'l',xlab='Gene number',ylab='SH test p-value',col=cols[1],xlim=c(0,200),ylim=c(0,1))
lines(sort(x$H2,decreasing = T),col=cols[2])
lines(sort(x$H4,decreasing = T),col=cols[3])
lines(sort(x$H5,decreasing = T),col=cols[4])
lines(sort(x$H7,decreasing = T),col=cols[5])
lines(sort(x$H9,decreasing = T),col=cols[6])
lines(sort(x$H12,decreasing = T),col=cols[7])
lines(sort(x$H15,decreasing = T),col=cols[8])
lines(sort(x$H16,decreasing = T),col=cols[9])
lines(sort(x$H17,decreasing = T),col=cols[10])

cols2=brewer.pal(7, "Dark2")
lines(sort(x$H3,decreasing = T),col=cols2[1])
lines(sort(x$H6,decreasing = T),col=cols2[2])
lines(sort(x$H8,decreasing = T),col=cols2[3])
lines(sort(x$H10,decreasing = T),col=cols2[4])
lines(sort(x$H11,decreasing = T),col=cols2[5])
lines(sort(x$H13,decreasing = T),col=cols2[6])
lines(sort(x$H14,decreasing = T),col=cols2[7])
abline(h = 0.95,lty=2)

x=read.table('KH_sum.includePandaceae.tsv',header = T)
plot(sort(x$H1,decreasing = T),pch=20,cex=0.4,type = 'l',xlab='Gene number',ylab='KH test p-value',col=cols[1],xlim=c(0,200),ylim=c(0,1))
lines(sort(x$H2,decreasing = T),col=cols[2])
lines(sort(x$H4,decreasing = T),col=cols[3])
lines(sort(x$H5,decreasing = T),col=cols[4])
lines(sort(x$H7,decreasing = T),col=cols[5])
lines(sort(x$H9,decreasing = T),col=cols[6])
lines(sort(x$H12,decreasing = T),col=cols[7])
lines(sort(x$H15,decreasing = T),col=cols[8])
lines(sort(x$H16,decreasing = T),col=cols[9])
lines(sort(x$H17,decreasing = T),col=cols[10])

cols2=brewer.pal(7, "Dark2")
lines(sort(x$H3,decreasing = T),col=cols2[1])
lines(sort(x$H6,decreasing = T),col=cols2[2])
lines(sort(x$H8,decreasing = T),col=cols2[3])
lines(sort(x$H10,decreasing = T),col=cols2[4])
lines(sort(x$H11,decreasing = T),col=cols2[5])
lines(sort(x$H13,decreasing = T),col=cols2[6])
lines(sort(x$H14,decreasing = T),col=cols2[7])
abline(h = 0.95,lty=2)
